### ISSUE TITLE
[feature] consolidate directory traversal checks

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/slsa-github-generator#1893]
We have a lot of directory traversal checks across all the actions and workflows. We should consolidate these checks into a some kind of library that can be re-used.

**Comment by asraa** [comment]
I believe the checks are in the tscommon file.ts now, so I think we have the library, and right now secure-upload-folder and download are using them

**Comment by ianlewis** [comment]
Right, we have some traversal checks done in bash which I'd like to get rid of. I just wish it was easier to write snippets of TS that can include reusable code and I didn't have to create a full blown action.

**Comment by ianlewis** [comment]
I have a couple places that do checks as a job step. Maybe we can also have a re-usable action that performs the checks?

**Comment by ianlewis** [comment]
> Let's start with adding a common lib for Go projects and have the container-based builder use it (+ pre-submits). Oak teams want to release it by end of April

I was mostly thinking about checks in the TS/bash code for workflows. I guess we also have some checking in the Go builders somewhere?

**Comment by laurentsimon** [comment]
Correct. Oak builder will go GA soon, so a good excuse before they announce it.

**Comment by laurentsimon** [comment]
@rbehjati mentioned that @asraa can work on the Go dir traversals - and have Oak builder use it - when she's back from OOO.

**Comment by ianlewis** [comment]
I think this could be done with two parts.
1. The existing function in tscommon (for TS actions)
2. A action that wraps the tscommon function (much like the `compute-sha256` action)
